custom_components/govee-ble-plugs/plugs.py

```diff
@@ -137,32 +136,136 @@ async def recv_handler(c, data):
                 elif data[0] == 0x33 and data[1] == 0x01:
                     on_set_state_ready.set()
 
-            await client.start_notify(self.RECV_CHARACTERISTIC_UUID, recv_handler)
+            await client.start_notify(self._RECV_CHARACTERISTIC_UUID, recv_handler)
 
             ba = bytearray([0x33, 0xB2]) + bytearray.fromhex(self._token).ljust(
                 17, b"\0"
             )
             ba.append(_sign_payload(ba))
-            await client.write_gatt_char(self.SEND_CHARACTERISTIC_UUID, ba)
+            await client.write_gatt_char(self._SEND_CHARACTERISTIC_UUID, ba)
             await on_auth_ready.wait()
 
-            ba = self.MSG_TURN_ON if new_state else self.MSG_TURN_OFF
-            await client.write_gatt_char(self.SEND_CHARACTERISTIC_UUID, ba)
+            await client.write_gatt_char(self._SEND_CHARACTERISTIC_UUID, msg)
             await on_set_state_ready.wait()
 
-            await client.stop_notify(self.RECV_CHARACTERISTIC_UUID)
+            await client.stop_notify(self._RECV_CHARACTERISTIC_UUID)
 
-            self._is_on = new_state
+            return True
         except Exception as e:
             _LOGGER.error("failed to set state: %s", e)
+            return False
         finally:
             if client is not None:
-                await client.disconnect()
+                try:
+                    await client.disconnect()
+                except EOFError:
+                    # sometimes this happens
```

Comment:
I don't remember this failing when I originally started working on this project. Maybe a kernel or bluez update broke something for me?